vpr/src/draw/search_bar.cpp

```diff
@@ -336,34 +406,111 @@ void warning_dialog_box(const char* message) {
     //Ensuring a valid type was selected
     if (!type) return;
     if (type[0] == '\0') return;
     std::string searchType(type);
-    //Setting active completion model to blockCompleter if type selected is block Name
+
+    /*
+     * If search type is name, connecting search bar to completion,
+     * and connecting completion to the appropriate model (blocks or nets)
+     * Additionally, visibility of key length setter is toggled by these changes
+     */
     if (searchType == "Block Name") {
-        GtkEntryCompletion* blockCompleter = GTK_ENTRY_COMPLETION(app->get_object("BlockNameCompleter"));
-        gtk_entry_set_completion(searchBar, blockCompleter);
-    } else { //If not, setting to null
+        gtk_entry_completion_set_model(completion, blockNames);
+        gtk_entry_set_completion(searchBar, completion);
+        gtk_widget_show(keyLengthSpinButton);
+        gtk_widget_show(keyLengthLabel);
+    } else if (searchType == "Net Name") {
+        gtk_entry_completion_set_model(completion, netNames);
+        gtk_entry_set_completion(searchBar, completion);
+        gtk_widget_show(keyLengthSpinButton);
+        gtk_widget_show(keyLengthLabel);
+    } else { //setting to null if option does not require auto-complete
+        gtk_widget_hide(keyLengthSpinButton);
+        gtk_widget_hide(keyLengthLabel);
         gtk_entry_set_completion(searchBar, nullptr);
     }
 }
 
 /**
  * @brief loads block names into gtk list store item used for completion
  * 
- * @param app ezgl application
+ * @param app ezgl application used for ui
  */
 void load_block_names(ezgl::application* app) {
     auto blockStorage = GTK_LIST_STORE(app->get_object("BlockNames"));
     auto& cluster_ctx = g_vpr_ctx.clustering();
+    auto& atom_ctx = g_vpr_ctx.atom();
     GtkTreeIter iter;
-    //Getting and storing all block names
     for (ClusterBlockId id : cluster_ctx.clb_nlist.blocks()) {
         gtk_list_store_append(blockStorage, &iter);
         gtk_list_store_set(blockStorage, &iter,
                            0, (cluster_ctx.clb_nlist.block_name(id)).c_str(), -1);
     }
+    for (AtomBlockId id : atom_ctx.nlist.blocks()) {
+        gtk_list_store_append(blockStorage, &iter);
+        gtk_list_store_set(blockStorage, &iter,
+                           0, (atom_ctx.nlist.block_name(id)).c_str(), -1);
+    }
+}
+
+/**
+ * @brief loads net names into gtk list store item used for completion
+ * 
+ * @param app ezgl application used for ui
+ */
+void load_net_names(ezgl::application* app) {
+    auto netStorage = GTK_LIST_STORE(app->get_object("NetNames"));
+    auto& atom_ctx = g_vpr_ctx.atom();
+    GtkTreeIter iter;
+    //Loading net names
+    for (AtomNetId id : atom_ctx.nlist.nets()) {
+        gtk_list_store_append(netStorage, &iter);
+        gtk_list_store_set(netStorage, &iter,
+                           0, (atom_ctx.nlist.net_name(id)).c_str(), -1);
+    }
+}
+
+/**
+ * @brief A new matching function used for the wildcard search options
```

Comment:
I'd explain this matching function searches for a substring anywhere in the name, rather than just a prefix (which is the default matcher).
I wouldn't call this wildcard since it doesn't do wildcards.
You say this will be slower --> if you have data on relative speed put that in the comment.